Poll

anewjean opened the pull request.

anewjean committed:
[WIP] Feat: Poll, Schedule 생성 기능 추가

- Poll Scheduler로 Poll을 생성 -> 이벤트 발행
- PollStatus별 Poll 조회
- Poll 생성 이벤트 발행 시 Schedule 생성
- pollId별 Schedule 리스트 조회
- schedule 투표

anewjean committed:
[WIP] Feat: Participation 기능 추가

- Schedule CONFIRMED 이벤트 발생하면 Participation을 영속화함
- useId별 Schedule 조회

anewjean committed:
[WIP] Feat: Participation 기능 추가

- Schedule CONFIRMED 이벤트 발생하면 Participation을 영속화함
- useId별 Schedule 조회

anewjean committed:
[WIP] Fix: 스케줄 확정 시 participation 저장 로직 추가

anewjean committed:
[WIP] Fix: 스케줄 확정 시 participation 저장 로직 추가

anewjean committed:
[WIP] Fix: 스케쥴 확정 로직 추가 등

changchanghwang commented:
base branch로 pr 날려주세요

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

changchanghwang committed:
Merge branch 'auth' of https://github.com/TeamOfAnts/Ant-server into poll

changchanghwang committed:
Merge branch 'poll' of https://github.com/TeamOfAnts/Ant-server into participation

changchanghwang committed:
fix: conflict

changchanghwang committed:
fix: conflict

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/poll/Poll.kt:
```diff
@@ -0,0 +1,46 @@
+    @Column(name = "end_at")
+    val endAt: Instant,
+
+    @Enumerated(EnumType.STRING)
```
wtf...

db에는 enum은 웬만하면 안쓰는게 좋습니다. -> 확장성이 너무 없어짐

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/poll/Poll.kt:
```diff
@@ -0,0 +1,46 @@
+@Table(name = "poll")
+data class Poll(
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
why auto inc?

anewjean committed:
Refactor: 모든 ApplicationException은 GlobalExceptionHandler가 처리하도록 수정

anewjean committed:
Refactor: 모든 ApplicationException은 GlobalExceptionHandler가 처리하도록 수정

anewjean committed:
Fix: 토큰 검증, 생성, 저장 로직은 TokenService가 처리하도록 변경

anewjean committed:
Fix: 토큰 검증, 생성, 저장 로직은 TokenService가 처리하도록 변경

anewjean committed:
Fix: refresh token 발급 시 불필요한 로직 삭제

anewjean committed:
Fix: refresh token 발급 시 불필요한 로직 삭제

anewjean committed:
Fix: 유저 업데이트 시 유저 엔티티 반환하지 않도록 수정

anewjean committed:
Fix: 유저 업데이트 시 유저 엔티티 반환하지 않도록 수정

anewjean committed:
Refactor: 분리되어있던 ApplicationException, Exception 처리 메서드 병합

sq

anewjean committed:
Refactor: 분리되어있던 ApplicationException, Exception 처리 메서드 병합

sq

anewjean committed:
Fix: 토큰 유효성에 대한 예외 처리 추가, JwtAuthenticationFilter 제외 path 추가

- JWT에 대한 verify는 TokenExpiredException, JWTDecodeException, SignatureVerificationException, AlgorithmMismatchException, InvalidClaimException 등 다양하지만 비즈니스 로직 상 TokenExpiredException와 그 외로 구분함
- JwtAuthenticationFilter에서 access token이 없는 경우 access token이 없어도 정상인 path (/health, /users/auth, /auth/refresh)는 Finally 통해 dofilter 후 permitAll() 되기를 기대했지만, permitAll 되기 전 에러가 발생하여 JwtAuthenticationFilter 자체를 통과하도록 수정함

anewjean committed:
Fix: 토큰 유효성에 대한 예외 처리 추가, JwtAuthenticationFilter 제외 path 추가

- JWT에 대한 verify는 TokenExpiredException, JWTDecodeException, SignatureVerificationException, AlgorithmMismatchException, InvalidClaimException 등 다양하지만 비즈니스 로직 상 TokenExpiredException와 그 외로 구분함
- JwtAuthenticationFilter에서 access token이 없는 경우 access token이 없어도 정상인 path (/health, /users/auth, /auth/refresh)는 Finally 통해 dofilter 후 permitAll() 되기를 기대했지만, permitAll 되기 전 에러가 발생하여 JwtAuthenticationFilter 자체를 통과하도록 수정함

anewjean committed:
Fix: import erorr

anewjean committed:
Fix: import erorr

anewjean committed:
Fix: 삭제한 custom 예외클래스 관련 Expired 코드 수정

anewjean committed:
Fix: 삭제한 custom 예외클래스 관련 Expired 코드 수정

anewjean committed:
Fix: base 브랜치 pull 관련 error 수정

anewjean committed:
Fix: base 브랜치 pull 관련 error 수정

anewjean committed:
Fix: @Scheduled 실행을 위한 @EnableScheduling 추가 등

anewjean committed:
Fix: @Scheduled 실행을 위한 @EnableScheduling 추가 등

anewjean committed:
Test: ScheduleScheduler 동작 시 updateScheduleStatus 실행 확인

anewjean committed:
Test: ScheduleScheduler 동작 시 updateScheduleStatus 실행 확인

anewjean committed:
Fix: auth 변경사항 반영

- JwtTokenManager
- TokenService

anewjean committed:
Fix: auth 변경사항 반영

- JwtTokenManager
- TokenService

anewjean committed:
Fix: Poll 조회 시 없으면 null 반환하도록 수정

anewjean committed:
Fix: Poll 조회 시 없으면 null 반환하도록 수정

anewjean committed:
Fix: 불필요한 import 삭제

anewjean committed:
Fix: 불필요한 import 삭제

anewjean committed:
Chore: 자신의 정보 조회 응답에서 providerId 제외

anewjean committed:
Merge branch 'participation' of https://github.com/TeamOfAnts/Ant-server into participation

anewjean committed:
Fix: conflict

anewjean committed:
Merge branch 'poll' into participation

anewjean committed:
Merge branch 'auth' into poll

anewjean committed:
Fix: poll 종료 시 poll status update로직 추가

anewjean committed:
Fix: poll.endAt을 스케쥴이 끝나는 날짜 -> 투표가 끝나는 날짜로 변경

anewjean committed:
Fix: Schedule에 대한 voter 추가 시 중복 투표 방지로직 추가

anewjean committed:
Refactor: 투표 시 응답 메세지 수정

anewjean committed:
Merge branch 'poll' into participation

anewjean committed:
Fix: import error

anewjean committed:
Fix: 스케쥴 확정시 CONFIRMED 상태의 스케쥴만 Participation 객체로 저장하도록 수정

anewjean committed:
Merge branch 'auth' into poll
